lib/clientv3/networkpolicy_e2e_test.go

```diff
@@ -468,4 +466,96 @@ var _ = testutils.E2eDatastoreDescribe("NetworkPolicy tests", testutils.Datastor
 			testWatcher4.Stop()
 		})
 	})
+
+	// These tests check that the names we use on the API properly round-trip.  In particular,
+	// k8s and OpenStack policies have special prefixes, which should be preserved.  Other
+	// names get stored with a prefix, for consistency but the API returns them without the
+	// prefix.
+	nameNormalizationTests := []TableEntry{
+		// OpenStack names should round-trip, including their prefix.
+		Entry("OpenStack policy", "ossg.default.group1"),
+		// As should normal names.
+		Entry("OpenStack policy", "foo-bar"),
+	}
+	if config.Spec.DatastoreType != "kubernetes" {
+		// Only test writing a knp-prefixed policy if we're not backed by KDD.  In KDD,
+		// the knp-prefixed policies are derived from k8s data so it doesn't make sense
+		// to write them through our API.
+		knpName := "knp.default.a-name"
+		nameNormalizationTests = append(nameNormalizationTests,
+			Entry("KDD policy", knpName),
+		)
+	}
+	DescribeTable("name round-tripping tests",
+		func(name string) {
+			c, err := clientv3.New(config)
+			Expect(err).NotTo(HaveOccurred())
+
+			be, err := backend.NewClient(config)
+			Expect(err).NotTo(HaveOccurred())
+			be.Clean()
```

Comment:
Is the backend client involved at all in these tests?  It's not clear to me if these tests are covering the thing that we most need, i.e. the mapping between API name and storage name for "ossg." NPs.  (But perhaps they are and I'm just not seeing it.)